### DISCUSSION TITLE
How to extract text excluding embedded text in images?

### DISCUSSION THREAD

**Discussion by MathieuChartier86** [jsvine/pdfplumber#728] [Ask for help with specific PDFs]
Hello,
I would like to be able to extract the text from the pages but without the extract_text() method doing the extraction in the images.
By default, the extract_text() method retrieves characters from plain text but also text from images. Unfortunately, extracting text from images is awful.
For example, here's a snippet of text that's been extracted from an image that better not be extracted: "<. Zô'-zX^ t i7^t^pi'r'r~n.^ Pc 'fold,»i'-<-<t!fX..y <^lMiS/e.j r'- yjx.—-\n., /f~ Z*/ z^r.-zzz*<*/*^ <r■?■>■» *z:z-Pe Z* *-•/?/»■**£>» ^m/y7<x-r^H<t^i-f Jc'%< en /'.x .^rPe <z?i,&l ------\nZ^rïX*'?,/ X'ir /•--zzrr-.^e- P< J't' ('Z^/'- er. /ej Æ-ï't.s.pZj .r/rn.t z>xxZléXPz/>rzX?«*-.zJ^e>Z"

I tried _page.filter(lambda obj: not (obj["object_type"] == "image"))_ but it doesn't change anything, I don't know what to do anymore.
If you can help me, I'm interested. Thanks in advance.

**Comment by MathieuChartier86** [comment]
Small precision: I have just noticed that the problem does not come from the images in reality. The problem comes from the fact that PdfPlumber considers some images as text, and this leads to many interpretation errors. How to prevent this?

**Comment by jsvine** [comment]
Hi @MathieuChartier86, and thanks for your interest in this library. There sounds like there may be some confusion here. `pdfplumber` does not extract any text from images, nor does it consider any images as text. Rather, my best guess (without seeing the actual PDF) is that the PDF itself contains text where those images appear. One way to test this: In your PDF viewer, click-drag your cursor across the whole page (that is, selecting all the text), copy the text, and then paste it into the text editor. Do you see the same issue?

**Reply by MathieuChartier86** [reply]
Thank you for your quick response (and congratulations on PdfPlumber, the library is awesome)!
Indeed, I find that I can select the text in the "images" I was describing. The PDFs come from an OCRization (OCR) so some real images have been understood as text.
I am creating a Dataset generator from a textual corpus (for my PhD), and the "noise" generated by the extraction of these "images" makes the task much more complicated. I would like to find a solution to limit as much as possible the extraction of these "weird" texts from ocerized images. Is there a solution with PdfPlumber to limit this behavior?

P.S.: I provide you with two screenshots from the PDFs I'm working on so that you can see the problem in a real case 
![OCR-PDF-Problem](https://user-images.githubusercontent.com/13598503/190470726-12945d5f-be98-4c9b-8b42-abce6a98a082.jpg)
![OCR-PDF-Problem-2](https://user-images.githubusercontent.com/13598503/190470732-16420413-6243-4291-8704-8d3d61663a74.jpg)
...

**Reply by jsvine** [reply]
Thank you for clarifying and for the examples, @MathieuChartier86. Do all of your PDFs begin as images that are then converted with OCR? If so, your options are limited. But if you want to distinguish between "born digital" PDFs and "converted image" PDFs, you could examine each `Page` object's `.images` property to determine the location of images (if any) on the page, and then exclude the relevant pages.

**Reply by MathieuChartier86** [reply]
Hello and thank you for your answer.
Unfortunately, I don't have a perfect solution to fix my problem. For the moment, I extract all the text, even that of the images coming from the OCR, then I use regular expressions to spot the "weird" texts and remove them. It's better than nothing. ;-)
